packages/react-dom/src/__tests__/ReactComponentLifeCycle-test.js

```diff
@@ -1057,70 +1057,69 @@ describe('ReactComponentLifeCycle', () => {
     });
   });
 
-  if (!require('shared/ReactFeatureFlags').disableModulePatternComponents) {
```

Comment:
These tests were implicitly depending on the fact that `disableModulePatternComponents` and `disableLegacyContext` always had the same value, so now we need to provide both.